drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -124,10 +127,38 @@ func (c *disruptionBudget) createPortworxPodDisruptionBudget(
 		return fmt.Errorf("failed to enumerate nodes: %v", err)
 	}
 
+	k8sNodeList := &v1.NodeList{}
+	err = c.k8sClient.List(context.TODO(), k8sNodeList)
+	if err != nil {
+		return err
+	}
+	k8sNodesStoragePodCouldRun := make(map[string]bool)
+	for _, node := range k8sNodeList.Items {
+		shouldRun, shouldContinueRunning, err := pxutil.CheckPredicates(&node, cluster, nil)
+		if err != nil {
+			return err
+		}
+		if shouldRun || shouldContinueRunning {
+			k8sNodesStoragePodCouldRun[node.Name] = true
+		}
+	}
+
 	storageNodesCount := 0
 	for _, node := range nodeEnumerateResponse.Nodes {
```

Comment:
add debug info output k8sNodesStoragePodCouldRun, nodeEnumerateResponse.Nodes and storageNodesCount.